rust/test/default.nix

```diff
@@ -1,12 +1,70 @@
 { holonix, pkgs }:
 let
-  name = "hc-rust-test";
-
-  script = pkgs.writeShellScriptBin name
+  hc-rust-test = pkgs.writeShellScriptBin "hc-rust-test"
   ''
   hc-rust-wasm-compile && HC_SIMPLE_LOGGER_MUTE=1 RUST_BACKTRACE=1 cargo test --all "$1" -- --test-threads=${holonix.rust.test.threads};
   '';
+
+  hc-rust-coverage-kcov = pkgs.writeShellScriptBin "hc-rust-coverage-kcov"
+  ''
+  # we need the kcov tool and curl for fetching the codecov.io uploader
+  nix-env -f https://github.com/NixOS/nixpkgs-channels/archive/nixos-19.09.tar.gz -iA kcov curl
+
+  # using cargo-make for executing kcov
+  cargo install cargo-make
```

Comment:
again... seems like maybe this should be part of the nix environment some magical way?

Reply:
we're fine sticking with this for ci task for now